File: Frontend/src/events/converters/player.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { PlayerDTO } from "@/events/dto/player.dto";
import type { Player } from "@/models";

export class PlayerConverter {
  public static convert(playerDto: DeepRequired<PlayerDTO>): Player {
    return {
      bits: playerDto.bits,
      mapId: playerDto.mapId,
      previousMapId: playerDto.previousMapId,
      seabedRoute: playerDto.seabedRoute,
      mapVariant: playerDto.mapVariant,
    };
  }
}

File: Backend/Events/Converters/PlayerConverter.cs
using Backend.Domain.Models;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class PlayerConverter
{
    public static PlayerDTO ToDTO(Player player)
    {
        return new PlayerDTO
        {
            Bits = player.Bits,
            Location = player.MapId,
            PreviousMapId = player.PreviousMapId,
            SeabedRoute = player.SeabedRoute,
            MapVariant = player.MapVariant
        };
    }
}
